scripts/cli_options.py

```diff
@@ -1,13 +1,71 @@
 #! /usr/bin/env python
 from __future__ import absolute_import, print_function
 import random
+from optparse import OptionParser, OptionValueError
+
+import jmclient.support
 
 """This exists as a separate module for two reasons:
 to reduce clutter in main scripts, and (TODO) refactor out
 options which are common to more than one script in a base class.
 """
 
-from optparse import OptionParser
+order_choose_algorithms = (
+    'random_under_max_order_choose', 'cheapest_order_choose',
+    'weighted_order_choose')
+
+
+def add_common_options(parser):
+    parser.add_option(
+        '-f',
+        '--txfee',
+        action='store',
+        type='int',
+        dest='txfee',
+        default=-1,
+        help='number of satoshis per participant to use as the initial estimate '
+        'for the total transaction fee, default=dynamically estimated, note that this is adjusted '
+        'based on the estimated fee calculated after tx construction, based on '
+        'policy set in joinmarket.cfg.')
+    parser.add_option('--fast',
+                      action='store_true',
+                      dest='fastsync',
+                      default=False,
+                      help=('choose to do fast wallet sync, only for Core and '
+                            'only for previously synced wallet'))
+    parser.add_option(
+        '-x',
+        '--maxcjfee',
+        type='float',
+        dest='maxcjfee',
+        nargs=2,
+        default=(0.01, 10000),
+        help='maximum coinjoin fee and bitcoin value the tumbler is '
```

Comment:
I'd prefer to just see one number here, because otherwise the default config might horribly fail if not changed by the user (e.g. if someone wishes to join a huger amount, like 100 btc or a very low amount like 0.01 btc).

We could get away with just using a "max relative fee" setting. And then calculate the resulting "max abs fee" depending on the actual cj amount on-the-fly.

Example:
Someone starts with defaults and wishes to join 0.01 btc with a 0.001 default (0.1%). Internally, without the user having to do so, the max absfee would be set to 1000 sat (0.1% of 0.01 btc) in this case.
If the same user would later wish to join 100 btc, without having to change anything, the max absfee would be 0.1 btc in that case.

Reply:
You have to exceed both limits to trigger the filter. This is exactly how the abort condition in `tumbler.py` currently works and so are the default values.

I think the motivation between having two values here is exactly in case you have a small cj amount. With just a relative check you might end up with an unrealistically low limit, rendering you unable to find any counterparties.

Reply:
In a 100btc cj the max relative fee allowed would be 1btc (1%) per maker according to the default values. I'm pretty sure that would actually allow all offers in the offer book. That said, the current defaults do seem pretty high.

Reply:
Hm. I dont get it then, it seems. A taker does not specify whether he wants absoffers or reloffers. Only makers do.

So the actual mechanism is:
If the *maker*'s offer is an absoffer, then the *taker* uses the max absoffer limit. If the *maker*'s offer is a reloffer, then the *taker* uses the max reloffer limit?

If so, then there's indeed not a problem (because only 1 of the limits is used per maker, not both).

I find it rather confusing though.

Reply:
I see where your confusion comes from. The fact that we have an absolute and relative max fee specified is independent of the offer type here. It's just two ways of determining a limit and then checking both (against all offers).

Reply:
Yes, 1% is definately too high. But wanted to talk about the general mechanism first before bikeshedding about actual values.

>  It's just two ways of determining a limit and then checking both (against all offers).

Then I'm confused again. So if the current default config would be used, this mechanism would discard any 100 btc maker (regardless whether abs or rel) which demands more than 10,000 sat for the join. Which, if i understand correctly, currently would make the join fail, because no maker offering that amount is accepting a cj fee that low (see here: https://joinmarket.me/ob/ ), even though the demanded fee is *FAR* below 1% (cheapest one at around 0.01%) .

Taken to the extremes (1000 btc join? 0.0001 btc join?) I believe a check against both values will make the defaults fail too often, because the 10,000 sat is too static. Checking only against a relative value would work always though (or can you give me an example when it wouldn't?).

I'll have a look at the actual code how the limits are used before my next reply, maybe we're just talking past each other.

Reply:
The `abs_max` is only relevant in the case of `cj_amount * rel_max < abs_max`. In that case we make sure a maker can always at least receive `abs_max` satoshi as cj fee. Otherwise only the limit determined by `cj_mount * rel_max` is relevant.

Reply:
The latest commit has some updated help text. Do you think that's better to understand?